internal/provisioner/utility/argo_utility.go

```diff
@@ -180,13 +180,17 @@ func (group utilityGroup) RemoveUtilityFromArgocd(gitClient git.Client) error {
 
 }
 
-func substituteValues(inputFilePath, outputFilePath string, replacements map[string]string) error {
+func substituteValues(inputFilePath, outputFilePath string, replacements map[string]string, logger log.FieldLogger) error {
 	inputFile, err := os.Open(inputFilePath)
 	if err != nil {
 		return errors.Wrap(err, "failed to open input file for argo utility substitution")
 	}
 	defer inputFile.Close()
 
+	if _, err = os.Stat(outputFilePath); err == nil {
+		logger.Debugf("Output file already exists, skipping substitution for %s", outputFilePath)
+		return nil
+	}
```

Comment:
Do we need to open the inputFile for any reason if the output file exists? If not this can be moved to the top of the function to be a bit more performant, but just a nit.